src/main/java/uk/gov/hmcts/reform/sscs/controller/CcdMideventCallbackController.java

```diff
@@ -142,6 +146,19 @@ public ResponseEntity<PreSubmitCallbackResponse<SscsCaseData>> ccdMidEventPrevie
         return ok(adjournCasePreviewService.preview(callback, DocumentType.DRAFT_ADJOURNMENT_NOTICE, userAuthorisation, false));
     }
 
+    @PostMapping(path = "/ccdMidEventAdjournDirectionDueDate", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<PreSubmitCallbackResponse<SscsCaseData>> ccdMidEventAdjournDirectionDueDate(
+            @RequestHeader(SERVICE_AUTHORISATION_HEADER) String serviceAuthHeader,
+            @RequestBody String message) {
+        Callback<SscsCaseData> callback = deserializer.deserialize(message);
+        log.info("About to start ccdMidEventAdjournDirectionDueDate callback `{}` received for Case ID `{}`", callback.getEvent(),
+                callback.getCaseDetails().getId());
+        authorisationService.authorise(serviceAuthHeader);
+
+
+        return ok(adjournCaseMidEventDueDateService.validateAdjournCaseDirectionsDueDateIsInFuture(callback));
```

Comment:
I think the callback response should be constructed here, and should just use the validation service to determine whether or not an error should be added